docs/dev/policy_documents/default-build-flags.md

```diff
@@ -0,0 +1,194 @@
+# Summary
+[summary]: #summary
+
+Habitat's core plans should provide a reliable base system on which
+Habitat users can build applications. To meet this goal, core-plans
+software written in the C programming language should use compiler and
+linker flags that provide a level of performance and security at least
+as good as other Linux distributions.
+
+# Motivation
+[motivation]: #motivation
+
+## Performance
+
+Consider our current build of `bc` attempting to compute Pi to 4096
+digits:
+
+```
+> time /hab/pkgs/core/bc/1.07.1/20190115012704/bin/bc -l <<< "scale=4096;4*a(1)"
+...SNIP...
+real	0m28.250s
+user	0m28.229s
+sys	0m0.016s
+```
+
+This same command recompiled with -O3:
+
+```
+> time /hab/pkgs/ssd/bc/1.07.1/20200427105511/bin/bc -l <<< "scale=4096;4*a(1)"
+...SNIP...
+real	0m10.950s
+user	0m10.960s
+sys	0m0.004s
+
+```
+
+This example was purposefully constructed to show a large speedup from
+optimization. However, I expect that the majority of software written
+in C in core-plans would benefit from ensuring we create optimized
+builds.
+
+Some software in core-plans already builds with optimizations, either
+because their build scripts add the required flags or because the plan
+we created does.
+
+However, it is common practice for distributions to set a default
+level of optimization for all builds. This is in part because some
+security features are also tied to having optimization enabled:
+
+For example,
+
+- [Omnibus: -02](https://github.com/chef/omnibus/blob/0baf021d62d9e61687e2d7efbac865a9640027c2/lib/omnibus/software.rb#L753)
+- [NixPkgs: -02 enabled when hardening is enabled (which is the default)](https://github.com/NixOS/nixpkgs/blob/4707dc6454904bee0bacd3a53829bcc76b4f0667/pkgs/build-support/cc-wrapper/add-hardening.sh#L41)
+- [Debian: -02](https://manpages.debian.org/testing/dpkg-dev/dpkg-buildflags.1.en.html)
+
+## Security
+
+Compilers have implemented a broad range of security
+mitigations. These include things like stack protectors that help
+prevent security bugs because of buffer overflows, automatic rewriting
+of unsafe libc functions into safer variants, and position independent
+executables to support address space layout randomization which makes
+various exploits harder to write.
+
+These mitigations are enabled via compiler and linker flags.  Most
+distributions have made such flags their default:
+
+https://fedoraproject.org/wiki/Changes/Harden_All_Packages
```

Comment:
I think you missed the `-` to make a list (it's hard to read otherwise)

Reply:
Thanks, fixed!